File: Diagrammatize.Design.Pattern/code/python/Flyweight/Sample/BigChar.py

class BigChar:
    def __init__(self, charname):
        self.charname = charname
        try:
            reader = open("big" + charname + ".txt")
            buf = []
            for line in reader:
                buf.append(line)
            reader.close()
            self.fontdata = "".join(buf)
        except Exception as e:
            self.fontdata = charname + "?"

    def print(self):
        print(self.fontdata, end="")

File: Diagrammatize.Design.Patterns/code/cxx/Flyweight/Sample/BigChar.cpp
#include "BigChar.hpp"
#include <iostream>
#include <fstream>
#include <sstream>

BigChar::BigChar(char charname_): charname(charname_) {
    std::ifstream reader(std::string("big") + charname+ ".txt");
    if (!reader) {
        fontdata = charname + std::string("?");
        return;
    }

    std::string line;
    std::stringstream buf;
    while (std::getline(reader, line)) {
        buf << line << "\n";
    }
    reader.close();
    fontdata = buf.str();
}

void BigChar::print() {
    std::cout << fontdata;
}
